keras_nlp/models/roberta.py

```diff
@@ -0,0 +1,298 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Roberta model configurable class, preconfigured versions, and task heads."""
+
+import tensorflow as tf
+from tensorflow import keras
+
+from keras_nlp.layers import TokenAndPositionEmbedding
+from keras_nlp.layers import TransformerEncoder
+
+
+def _roberta_kernel_initializer(stddev=0.02):
+    return keras.initializers.TruncatedNormal(stddev=stddev)
+
+
+class Roberta(keras.Model):
+    """Robustly optimized Bert pretraining approach.
+
+    This network implements a bi-directional Transformer-based encoder as
+    described in ["RoBERTa: A Robustly Optimized BERT Pretraining Approach"](https://arxiv.org/abs/1907.11692).
+    It includes the embedding lookups and transformer layers, but not include
+    the masked language model network.
+
+    This class gives a fully configurable Roberta model with any number of
+    layers, heads, and embedding dimensions. For specific roberta architectures
+    defined in the paper, see for example `keras_nlp.models.RobertaBase`.
+
+    Args:
+        vocabulary_size: Int. The size of the token vocabulary.
+        num_layers: Int. The number of transformer layers.
+        num_heads: Int. The number of attention heads for each transformer.
+            The hidden size must be divisible by the number of attention heads.
+        hidden_dim: Int. The size of the transformer encoding layer.
+        intermediate_dim: Int. The output dimension of the first Dense layer in
+            a two-layer feedforward network for each transformer.
+        dropout: Float. Dropout probability for the Transformer encoder.
+        max_sequence_length: Int. The maximum sequence length that this encoder
+            can consume. If None, `max_sequence_length` uses the value from
+            sequence length. This determines the variable shape for positional
+            embeddings.
+        start_token_index: Int. Index of <s> token in the vocabulary. Equivalent
+            to [CLS] in BERT.
+        name: String, optional. Name of the model.
+        trainable: Boolean, optional. If the model's variables should be
+            trainable.
+
+    Example usage:
+    ```python
+    # Randomly initialized Roberta encoder
+    encoder = keras_nlp.models.Roberta(
+        vocabulary_size=50265,
+        num_layers=12,
+        num_heads=12,
+        hidden_dim=768,
+        intermediate_dim=3072,
+        max_sequence_length=12
+    )
+
+    # Call encoder on the inputs.
+    input_data = {
+        "input_ids": tf.random.uniform(
+            shape=(1, 12), dtype=tf.int64, maxval=50265),
+        "input_mask": tf.constant(
+            [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 0], shape=(1, 12)),
+    }
+    output = encoder(input_data)
+    ```
+    """
+
+    def __init__(
+        self,
+        vocabulary_size,
+        num_layers,
+        num_heads,
+        hidden_dim,
+        intermediate_dim,
+        dropout=0.1,
```

Comment:
This question goes beyond this specific PR, but why do our models use this specific dropout value, as opposed to 0.? Either you use the model for inference, and there's no dropout, or you're training / fine tuning, in which case you almost certainly want to configure the value yourself.

Reply:
For BERT, this is the value used in the paper so it seemed helpful to offer a "best practice" default. I'm not sure about Roberta

Reply:
Yeah, it seems nice to set up with the same dropout used for pretraining/finetuning by default by the original authors where possible. Let's just double check in the fairseq code if we have the right constant here.